crates/kitsune_p2p/kitsune_p2p/tests/common/test_keystore.rs

```diff
@@ -0,0 +1,40 @@
+use std::sync::Arc;
+
+use futures::lock::Mutex;
+use kitsune_p2p_types::dependencies::lair_keystore_api::{self, LairClient, LairResult};
+use tokio::runtime::Handle;
+use tx5::deps::{hc_seed_bundle::PwHashLimits, sodoken};
+
+/// Construct a new TestKeystore with the new lair api.
+pub async fn spawn_test_keystore() -> LairResult<Arc<Mutex<LairClient>>> {
+    // in-memory secure random passphrase
+    let passphrase = sodoken::BufWrite::new_mem_locked(32)?;
+    sodoken::random::bytes_buf(passphrase.clone()).await?;
+
+    // in-mem / in-proc config
+    let config = Arc::new(
+        PwHashLimits::Minimum
+            .with_exec(|| {
+                lair_keystore_api::config::LairServerConfigInner::new("/", passphrase.to_read())
+            })
+            .await?,
+    );
+
+    // the keystore
+    let keystore = lair_keystore_api::in_proc_keystore::InProcKeystore::new(
```

Comment:
Pretty sure if this is dropped the store itself will no longer exist and the client will fail. If it works to *not* persist this, there may be a memory leak in the Lair code.

Reply:
Hard to be sure looking at the code, but it looks like the client might be holding onto a reference counted server handle which would be enough to keep Lair alive. The InProcKeystore will be dropped yes, but that doesn't have anything we need again, just the config and passphrase.

I can make the code hold the keystore if this isn't intentional and shouldn't work?

Reply:
Hmm, well cool, if it works, I guess let's leave as-is for now.